components/prf/hmac-sha256/src/prf.rs

```diff
@@ -246,24 +227,45 @@ where
     E: Memory + Load + Execute + Decode + Send,
 {
     #[instrument(level = "debug", skip_all, err)]
-    async fn preprocess(&mut self, pms: ValueRef) -> Result<SessionKeys, PrfError> {
+    async fn setup(&mut self, pms: ValueRef) -> Result<SessionKeys, PrfError> {
         let State::Initialized = self.state.take() else {
             return Err(PrfError::state("PRF not in initialized state"));
         };
 
-        let visibility = match self.config.role {
-            Role::Leader => Visibility::Private,
-            Role::Follower => Visibility::Blind,
+        let thread = &mut self.thread_0;
+
+        let randoms = Randoms {
+            client_random: thread.new_input::<[u8; 32]>(
+                "client_finished",
+                match self.config.role {
+                    Role::Leader => Visibility::Private,
```

Comment:
Can we add here an explanatory comment lest we forget. Why are we making the client_random private? I'm currently not sure myself.
Is it A) 
to add entropy to the handshake to enable domain name privacy during attestation (still need to investigate if server signature alone has enough entropy). If not, then is it B) to simplify the code logic and to avoid sending the `client_random` to the follower in the online phase?